tool/src/org/antlr/v4/codegen/target/GoTarget.java

```diff
@@ -54,6 +57,26 @@ protected void addBadWords() {
 	}
 
 	@Override
+	protected void genFile(Grammar g, ST outputFileST, String fileName) {
+		super.genFile(g, outputFileST, fileName);
+		gofmt(fileName); // consider commenting out when debugging
+	}
+
+	private void gofmt(String fileName) {
+		// If the generated source file is for any reason invalid, gofmt will not change the file and generate output on
+		// stderr. The output of stderr will be lost, gofmt may block asynchronously if the output buffer is not
+		// drained, but in that case gofmt will be killed as soon the JVM terminates. In normal operations the gofmt
+		// process will not get killed prematurely and not create a race condition with the lifecycle of the JVM.
```

Comment:
FWIW, I found [this](http://stackoverflow.com/a/1130774/1110390), which seems pretty simple:

	try {
	    ProcessBuilder pb = new ProcessBuilder("/bin/chmod", "777", path);    
	    pb.redirectErrorStream(true); // merge stdout, stderr of process

	    Process p = pb.start();
	    InputStreamReader isr = new  InputStreamReader(p.getInputStream());
	    BufferedReader br = new BufferedReader(isr);

	    String lineRead;
	    while ((lineRead = br.readLine()) != null) {
	        // swallow the line, or print it out - System.out.println(lineRead);
	    }

	    int rc = p.waitFor();
	    // TODO error handling for non-zero rc
	}
	catch (IOException e) {
	    e.printStackTrace(); // or log it, or otherwise handle it
	}
	catch (InterruptedException ie) {
	    ie.printStackTrace(); // or log it, or otherwise handle it
	}